be1-go/internal/popserver/handler/query_test.go

```diff
@@ -0,0 +1,642 @@
+package handler
+
+import (
+	"encoding/json"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+	"popstellar/crypto"
+	"popstellar/internal/popserver/config"
+	"popstellar/internal/popserver/database"
+	"popstellar/internal/popserver/database/repository"
+	"popstellar/internal/popserver/generatortest"
+	"popstellar/internal/popserver/state"
+	"popstellar/internal/popserver/types"
+	"popstellar/message/query/method"
+	"popstellar/message/query/method/message"
+	"popstellar/network/socket"
+	"testing"
+)
+
+func Test_handleQuery(t *testing.T) {
```

Comment:
This file is pretty long. Since you have a test for each of the JSON-RPC level RPCs, and distinct logic for each of them as well, you could also move this whole thing in its own package and split out in smaller files for each RPC. It'd make it easier to maintain and quickly find the relevant bit of code.